src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,29 +17,34 @@ public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes a person.\n"
+            + "Parameters: "
+            + PREFIX_STUDENTNUMBER + "STUDENT NUMBER\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_STUDENTNUMBER + "A0245234A";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private final StudentNumber targetStudentNumber;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(StudentNumber targetStudentNumber) {
+        this.targetStudentNumber = targetStudentNumber;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (!this.targetStudentNumber.isValid()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_NUMBER);
+        }
+
+        if (!model.hasPerson(new Person(targetStudentNumber))) {
```

Comment:
Is it possible to check if model has person using Student number without creating a Person object?